presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -133,7 +136,9 @@ public void checkCanShowTablesMetadata(ConnectorSecurityContext context, String
     @Override
     public Set<SchemaTableName> filterTables(ConnectorSecurityContext context, Set<SchemaTableName> tableNames)
     {
-        return tableNames;
+        return tableNames.stream()
+                .filter(tableName -> isSchemaOwner(context, tableName.getSchemaName()) || checkTableAnyPermission(context, tableName))
```

Comment:
Please add a comment why we have `isSchemaOwner(context, tableName.getSchemaName()) ||` here.

Reply:
sure